services/jira/jira-sprint.service.js

```diff
@@ -33,74 +56,57 @@ module.exports = class JiraSprint extends LegacyService {
           host: 'jira.spring.io',
           sprintId: '94',
         },
-        staticPreview: {
-          label: 'completion',
-          message: '96%',
+        staticPreview: this.render({
+          percentComplete: 96,
           color: 'orange',
-        },
+        }),
         documentation,
+        keywords: ['jira', 'sprint', 'issues'],
       },
     ]
   }
 
-  static registerLegacyRouteHandler({ camp, cache }) {
-    camp.route(
-      /^\/jira\/sprint\/(http(?:s)?)\/(.+)\/([^/]+)\.(svg|png|gif|jpg|json)$/,
-      cache((data, match, sendBadge, request) => {
-        const protocol = match[1] // eg, https
-        const host = match[2] // eg, jira.spring.io
-        const sprintId = match[3] // eg, 94
-        const format = match[4] // eg, png
-
-        const options = {
-          method: 'GET',
-          json: true,
-          uri: `${protocol}://${host}/rest/api/2/search?jql=sprint=${sprintId}%20AND%20type%20IN%20(Bug,Improvement,Story,"Technical%20task")&fields=resolution&maxResults=500`,
-        }
-        if (serverSecrets && serverSecrets.jira_username) {
-          options.auth = {
-            user: serverSecrets.jira_username,
-            pass: serverSecrets.jira_password,
-          }
-        }
+  async handle({ protocol, host, path, sprintId }) {
+    let url = `${protocol}://${host}`
+    if (path) {
+      url += `/${path}`
+    }
+    // Atlassian Documentation: https://developer.atlassian.com/cloud/jira/platform/rest/v2/#api-group-Search
+    // There are other sprint-specific APIs but those require authentication. The search API
+    // allows us to get the needed data without being forced to authenticate.
+    url += '/rest/api/2/search'
+    const options = {
+      qs: {
+        jql: `sprint=${sprintId} AND type IN (Bug,Improvement,Story,"Technical task")`,
```

Comment:
I added the above comment on `url` to indicate the usage of `search` over `sprint` API, and I totally get why it is implemented that way. However, I could envision some potential issues due the issue type clause that we have to put into the jql query. Jira is used by some teams at my day job, and we've created many custom issue types (like 12387123897612387123 of them 😄)  that wouldn't be accurately reflected by this query.

IMO we shouldn't try to solve that in this PR, but it may be worth revisiting at some point